### PR TITLE
Ameerul /P2PS-1109 Refactor Delete Payment Method Modal

### PR DESCRIPTION
**Pull request by ameerul-deriv**
## Changes:

Please provide a summary of the change.
- Refactored DeletePaymentMethodModal and added test cases
- Added new modal ConfirmDeletePaymentMethodModal and removed instances of it from stores

### Screenshots:

Please provide some screenshots of the change.
Screenshot 2023-08-02 at 11 17 21 AM
Screenshot 2023-08-02 at 10 57 48 AM

### CODE REVIEW

File: packages/p2p/src/components/modal-manager/modals/confirm-delete-payment-method-modal/__tests__/confirm-delete-payment-method-modal.spec.tsx [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+    });
+
+    it('should render the ConfirmDeletePaymentMethodModal', () => {
+        render(<ConfirmDeletePaymentMethodModal />);
+
+        expect(screen.getByText('Delete test name?')).toBeInTheDocument();
+        expect(screen.getByText('Are you sure you want to remove this payment method?')).toBeInTheDocument();
+        expect(screen.getByRole('button', { name: 'Yes, remove' })).toBeInTheDocument();
+        expect(screen.getByRole('button', { name: 'No' })).toBeInTheDocument();
+    });
```
Comment (farrah-deriv) [inline packages/p2p/src/components/modal-manager/modals/confirm-delete-payment-method-modal/__tests__/confirm-delete-payment-method-modal.spec.tsx:51]: please add checks if the user confirms to delete the payment method or not

File: packages/p2p/src/components/modal-manager/modals/delete-payment-method-error-modal/__tests__/delete-payment-method-error-modal.spec.tsx [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+    afterAll(() => {
+        document.body.removeChild(el_modal);
+    });
+
+    it('should render the DeletePaymentMethodErrorModal', () => {
+        render(<DeletePaymentMethodErrorModal />);
+
+        expect(screen.getByText('That payment method cannot be deleted')).toBeInTheDocument();
+        expect(screen.getByText('error message')).toBeInTheDocument();
+        expect(screen.getByRole('button', { name: 'Ok' })).toBeInTheDocument();
```
Comment (farrah-deriv) [inline packages/p2p/src/components/modal-manager/modals/delete-payment-method-error-modal/__tests__/delete-payment-method-error-modal.spec.tsx:46]: please add a check for the button too

File: packages/p2p/src/components/modal-manager/modals/confirm-delete-payment-method-modal/confirm-delete-payment-method-modal.tsx [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+            has_close_icon={false}
+            title={
+                <Text color='prominent' weight='bold'>
+                    <Localize
+                        i18n_default_text='Delete {{payment_method_name}}?'
+                        values={{
+                            payment_method_name:
+                                my_profile_store?.payment_method_to_delete?.fields?.bank_name?.value ||
+                                my_profile_store?.payment_method_to_delete?.fields?.name?.value ||
+                                my_profile_store?.payment_method_to_delete?.display_name,
```
Comment (farrah-deriv) [inline packages/p2p/src/components/modal-manager/modals/confirm-delete-payment-method-modal/confirm-delete-payment-method-modal.tsx]: please destructure this

**Review by farrah-deriv — CHANGES_REQUESTED** [review]

File: packages/p2p/src/pages/my-profile/payment-methods/payment-methods-list/payment-methods-list.jsx [hunk 4]
```diff
@@ -137,48 +129,6 @@ const PaymentMethodsList = () => {
                     </div>
                 </MobileFullPageModal>
             </MobileWrapper>
-            <Modal
```
Comment (farhan-nurzi-deriv) [inline packages/p2p/src/pages/my-profile/payment-methods/payment-methods-list/payment-methods-list.jsx:140]: Nice, why this modal is here 😆

**Review by farhan-nurzi-deriv — APPROVED** [review]

File: packages/p2p/src/pages/my-profile/payment-methods/payment-methods-list/payment-methods-list.jsx (hunk 4 above)
Comment (ameerul-deriv) [reply packages/p2p/src/pages/my-profile/payment-methods/payment-methods-list/payment-methods-list.jsx:140]: yeah there are some modals that havent been moved out yet 👀

File: packages/p2p/src/components/modal-manager/modals/delete-payment-method-error-modal/__tests__/delete-payment-method-error-modal.spec.tsx [hunk 5]
```diff
@@ -0,0 +1,57 @@
+import React from 'react';
+import { screen, render } from '@testing-library/react';
+import { useModalManagerContext } from 'Components/modal-manager/modal-manager-context';
+import { useStores } from 'Stores/index';
+import DeletePaymentMethodErrorModal from '../delete-payment-method-error-modal';
+import userEvent from '@testing-library/user-event';
```
Comment (nada-deriv) [inline packages/p2p/src/components/modal-manager/modals/delete-payment-method-error-modal/__tests__/delete-payment-method-error-modal.spec.tsx]: can we reorder this?

**Review by nada-deriv — APPROVED** [review]

**Review by niloofar-deriv — APPROVED** [review]
LGTM :+1:

**Review by farrah-deriv — APPROVED** [review]

**Pull request merged**